app/models/system_tag.rb

```diff
@@ -0,0 +1,35 @@
+class SystemTag < ApplicationRecord
+  belongs_to :parent, optional: true, class_name: "SystemTag", inverse_of: :system_tags
+  belongs_to :organization, optional: true
+  has_many :system_tags, as: :parent, class_name: "SystemTag", foreign_key: :parent_id, inverse_of: :parent
```

Comment:
Nestable tags feel counter-intuitive to me. Seems contrary to the concept of tagging, which is in some ways designed to get away from categorization. I think this 'tag grouping' is a very specific UI treatment that one of the Michigan groups wants and i'm not sure we should design the models to match that.

In the initial implementation, the tagging system didn't rely on a storing tags in the DB. I'm not partial to that, but either way, once we start accommodating user-defined tags, we're going to have to manage both sets, and i think a flat structure could be easier to work with.

What if we went with a 'category' field on system tags instead of nesting?

Reply:
This hierarchical taxonomy is needed by multiple groups. These tags are used to populate dropdowns on forms and nested filters on things like the Share page, but anything _actually_ tagged, would go in the taggable table in the db.

Reply:
I don't love forcing category and one level of nesting, but am open to it -- at least to start. We'd need another table tho, to store Categories so they can set their own?